### PR TITLE
Makefile/release targets

### PR DESCRIPTION
**Pull request by phoinixgrr**
#### Summary

[feat] Makefile addition of build target https://github.com/mattermost/rtcd/pull/47/commits/d93452664e8f6cf0fb5abc3ec363edadca85451a
  
  - build as a catch all target for all build related jobs
  - go-build-docker as a non workstation binded build target, thus building inside a predefined controll golang container

[feat] Makefile addition of github-release https://github.com/mattermost/rtcd/pull/47/commits/58fb8ad0c1f29bab716ae43d35df28706f74ba4d
  - Publishes a Github Release (only when triggered from a tag)
  - Populates the release notes with the changes introduced
  - Uploads build artifacts

[fix] Variables declaration order https://github.com/mattermost/rtcd/pull/47/commits/660c48f22b496d607fe4f6a4c2a1993cbbab6d45

#### Ticket Link

Ticket: https://mattermost.atlassian.net/browse/DOPS-908

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -287,6 +296,22 @@ go-doc: ## to generate documentation
 	$(AT)$(GO) run ./scripts/env_config.go ./docs/env_config.md || ${FAIL}
 	@$(OK) Generating Documentation
 
+.PHONY: github-release
+github-release: ## to publish a release and relevant artifacts to GitHub
+	@$(INFO) Generating github-release http://github.com/$(GITHUB_ORG)/$(GITHUB_REPO)/releases/tag/$(APP_VERSION) ...
+ifeq ($(shell echo $(APP_VERSION) | egrep '^v([0-9]+\.){0,2}(\*|[0-9]+)'),)
+	$(error "We only support releases from semver tags")
```
Comment (pfltdv) [inline Makefile:303]: :heart: That is sweet.

File: Makefile [hunk 2]
```diff
@@ -26,6 +26,7 @@ DOCKER_IMAGE_GO         += "golang:${GO_VERSION}@sha256:79138c839452a2a9d767f0bb
 DOCKER_IMAGE_GOLINT     += "golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b"
 DOCKER_IMAGE_DOCKERLINT += "hadolint/hadolint:v2.9.2@sha256:d355bd7df747a0f124f3b5e7b21e9dafd0cb19732a276f901f0fdee243ec1f3b"
 DOCKER_IMAGE_COSIGN     += "bitnami/cosign:1.8.0@sha256:8c2c61c546258fffff18b47bb82a65af6142007306b737129a7bd5429d53629a"
+DOCKER_IMAGE_GH_CLI     += "registry.internal.mattermost.com/images/build-ci:3.16.0@sha256:f6a229a9ababef3c483f237805ee4c3dbfb63f5de4fbbf58f4c4b6ed8fcd34b6"
```
Comment (pfltdv) [inline Makefile:52]: Do we have any plans to execute this script outside of our vpn? For contributors and  outside VPN that image will not be available.

Comment (phoinixgrr) [reply Makefile:52]: Yes thought of it, but unfortunately there is no official supported github docker container published. 
Therefore I've opted for using our image for security reasons.

Comment (pfltdv) [reply Makefile:52]: Cool thank you, with that intention nothing from me atm.

**Review by pfltdv — APPROVED** [review]
:100: Nice work @phoinixgrr

**Pull request merged**